File: backend/models/cs_class.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

type CSClass struct {
	ID       primitive.ObjectID   `json:"_id,omitempty" bson:"_id,omitempty"`
	Name     string               `json:"name"`
	Queues   []primitive.ObjectID `json:"queues"`
	IsActive bool                 `json:"isActive"`
	Semester string               `json:"semester"`
	Year     int                  `json:"year"`
}

func (c *CSClass) TableName() string {
	return "cs_classes"
}

File: frontend/src/types/csClass.ts
/*
 * csClass.ts
 *
 * This file contains a bunch of types and enums for the CSClass data.
 *
 * This file is used to define the types and enums for the CSClass data
 */
import { ObjectId } from "mongodb";
import { TaQueue } from "./taQueue";

// This is for the CSClass data
//
// It is for all CSClass objects
export interface CSClass {
    _id?: ObjectId;
    name: string;
    activeQueue: ObjectId;
    queues: ObjectId[];
    isActive: boolean;
    semester: string;
    year: number;
}

// This is the parameters for creating CS Classes
export type CreateCSClassParams = {
    name: string | null;
    semester: string | null;
    year: number | null;
};

// This enum is for the CSClass data
//
// Used mostly for dropdowns 
export enum SEMESTER_NAMES {
    FALL = "Fall",
    SPRING = "Spring",
}

// This is what the hook for creating TA queues returns
export type CreateTaQueueResponse = {
    taqueue: TaQueue;
};
